src/components/tree-view.tsx

```diff
@@ -129,7 +138,23 @@ const TreeItem = ( {
 				{ ! node.loading && node.children && ! node.hideExpandButton && (
 					<button
 						aria-label={ expanded ? __( 'Collapse' ) : __( 'Expand' ) }
-						onClick={ () => onPatchNode( node.id, { expanded: ! expanded } ) }
+						onClick={ async () => {
+							if (
+								! expanded &&
+								onExpand &&
+								node.children &&
+								node.children.length === 0 &&
```

Comment:
Having `node.children` separately looks confusing, since it makes feeling that it's independent check, but actually it's just to check the `length` property safely. So let's use optional chaining:
```suggestion
								node.children?.length === 0 &&
```

Reply:
done https://github.com/Automattic/studio/pull/1596/commits/bf3877591737f624c393f0576cfb73b073e0bba7